server/src/main/java/io/crate/expression/udf/UserDefinedFunctionService.java

```diff
@@ -240,4 +263,37 @@ public FunctionProvider buildFunctionResolver(UserDefinedFunctionMetadata udf) {
 
         return new FunctionProvider(signature, (s, args) -> scalar);
     }
+
+    private void validateFunctionIsNotInUseByGeneratedColumn(String schema,
+                                                             String functionName,
+                                                             UserDefinedFunctionsMetadata functions,
+                                                             Schemas schemas) {
+        // The iteration of schemas/tables must happen on the node context WITHOUT the UDF already removed.
+        // Otherwise the lazy table factories will already fail while evaluating generated functions.
+        // To avoid that, a copy of the node context with the removed UDF function is used on concrete expression evaluation.
+        var nodeCtx = new NodeContext(this.nodeCtx.functions().copyOf());
+        updateImplementations(schema, functions.functionsMetadata().stream(), nodeCtx);
+
+        for (var schemaInfo : schemas) {
```

Comment:
Shouldn't this operate on the `currentState` of the `ClusterStateUpdateTask` ?

`Schemas` is updated via `clusterChanged` events and I think there could be some race conditions - e.g. UDF being dropped while concurrently a new table is created that uses it.

Reply:
right, also recognized just now. will come up with a new approach.

Reply:
I've pushed a fixup solving this.

Reply:
good point, was even simple to do. pushed another fixup (this also reverts the commit before).